File: Math/Max Points on a Line/MaxPoints.cpp
class Solution {
public:
    int maxPoints(vector<vector<int>>& points) {
        int counter = 1;
        if (points.size() < 2) {
            return 1;
        }
        for (int i = 0; i < points.size(); ++i) {
            unordered_map<double, int> slope;
            for (int j = i + 1; j < points.size(); ++j) {
                int y = points[j][1] - points[i][1];
                int x = points[j][0] - points[i][0];
                if (x != 0) {
                    slope[static_cast<double>(y) / x] = 1 + slope[static_cast<double>(y) / x];
                } else {
                    slope[numeric_limits<double>::infinity()] = 1 + slope[numeric_limits<double>::infinity()];
                }
            }
            for (const auto& kv : slope) {
                counter = max(counter, kv.second);
            }
        }        
        return counter + 1;
    }
};

File: Math/Max Points on a Line/max_points.py
class Solution:
    def maxPoints(self, points: List[List[int]]) -> int:
        if len(points) <= 2:
            return len(points)

        result = 0
        for p1 in points:
            collinear_points = {}
            for p2 in points:
                if p1 == p2:
                    continue

                slope = 0
                if p2[0] == p1[0]:
                    slope = float("inf")
                else:
                    slope = (p2[1] - p1[1]) / (p2[0] - p1[0])

                collinear_points[slope] = collinear_points.get(slope, 0) + 1

                if collinear_points[slope] > result:
                    result = collinear_points[slope]

        return result + 1
